app/LinearAlgebra.hs

```diff
@@ -0,0 +1,4 @@
+import Math.NumberTheory.Primes.Factorisation.LinearAlgebra
+
+main :: IO ()
+main = print $ testLinearSolver 100 0.1
```

Comment:
I think you vastly overestimate sparsity of matrices, relevant to quadratic sieve here. This may be the reason of your algorithm being slow. 

For instance, `arithmoi-quadratic-sieve` generates 5621 equations over 1489 primes, where only 39701 elements are non-zero. This correponds to sparsity 0.005.